plugins/field-slider/test/field_test_helpers.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const assert = require('chai').assert;
+const Blockly = require('blockly');
+
+
+/**
+ * @typedef {{
+ *            title:string,
+ *            value:*
+ *            expectedValue:?
+ *          }}
+ */
+let Run;
+
+/**
+ * Assert a field's value is the same as the expected value.
+ * @param {Blockly.Field} field The field.
+ * @param {*} expectedValue The expected value.
+ * @param {string=} opt_expectedText The expected text.
+ */
+function assertFieldValue(field, expectedValue, opt_expectedText) {
+  const actualValue = field.getValue();
+  const actualText = field.getText();
+  opt_expectedText = opt_expectedText || String(expectedValue);
+  assert.equal(actualValue, expectedValue, 'Value');
+  assert.equal(actualText, opt_expectedText, 'Text');
+}
+
+/**
+ * Runs provided creation test cases.
+ * @param {Array<Run>} runs The test cases to run.
+ * @param {function(T, Run)} assertion The assertion to use.
+ * @param {function(new:T,Run):T} creation The creation method to use.
+ * @template {!Blockly.Field} T
```

Comment:
Any reason for the template here? All fields inherit from Blockly.Field, and simply ``function(Blockly.Field, Run)`` should work.

Reply:
I was trying to get fancy and say that the type used is a subtype of Field.
It seemed to be more correct to use generics as I could use the same type in all the params.
I'm fine with changing back to Blockly.Field as I'm having some trouble recalling any specific benefits to this.